actions/pr-quality-check/action.yml

```diff
@@ -0,0 +1,1033 @@
+        echo "- **Warnings**: $TOTAL_WARNINGS" >> quality_results.md
+        echo "" >> quality_results.md
+
+        TOTAL_ISSUES=$((TOTAL_ERRORS + TOTAL_WARNINGS))
+        if [ $TOTAL_ISSUES -eq 0 ]; then
+          echo "✅ **All checks passed!** Your code meets the quality standards." >> quality_results.md
+        else
+          if [ $TOTAL_ERRORS -gt 0 ]; then
+            echo "❌ **Please address the issues above before merging.**" >> quality_results.md
+          else
+            echo "⚠️ **Some warnings found.** Consider addressing them to improve code quality." >> quality_results.md
+          fi
+        fi
+
+        echo "" >> quality_results.md
+        echo "---" >> quality_results.md
+        echo "*Generated by PR Quality Check* 🤖 - Comment updated with every commit (last refresh: ${HEAD_SHA_SHORT}) " >> quality_results.md
+
+        echo "TOTAL_CHANGED_FILES=$TOTAL_CHANGED_FILES" >> $GITHUB_ENV
+        echo "TOTAL_ERRORS=$TOTAL_ERRORS" >> $GITHUB_ENV
+        echo "TOTAL_WARNINGS=$TOTAL_WARNINGS" >> $GITHUB_ENV
+        echo "TOTAL_FILES_CHECKED=$TOTAL_FILES_CHECKED" >> $GITHUB_ENV
+
+    - name: Post quality check results
+      if:
+        env.SHOULD_RUN == 'true' && env.HAS_MATCHING_CHANGES == 'true' &&
+        inputs.post-comment == 'true'
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.gh-token }}
+        PR_NUMBER: ${{ github.event.pull_request.number }}
+        REPO: ${{ github.repository }}
+      run: |
+        echo "💬 Posting quality check results to PR #$PR_NUMBER"
+
+        # Check if we already have a comment from this action
+        EXISTING_COMMENT=$(gh api repos/$REPO/issues/$PR_NUMBER/comments --jq '.[] | select(.body | contains("Generated by PR Quality Check")) | .id' | head -1)
+
+        if [ ! -z "$EXISTING_COMMENT" ]; then
+          echo "Updating existing comment (ID: $EXISTING_COMMENT)"
+          gh api repos/$REPO/issues/comments/$EXISTING_COMMENT \
+            --method PATCH \
+            --field body="$(cat quality_results.md)" \
+            --silent
+          echo "✅ Updated existing quality check comment!"
+        else
+          echo "Creating new comment"
+          gh pr comment $PR_NUMBER --body-file quality_results.md
+          echo "✅ Posted quality check results!"
+        fi
+
+        # Output summary
+        echo "📊 Quality Check Summary:"
+        echo "  Files changed: $TOTAL_CHANGED_FILES"
+        echo "  Files analyzed: $TOTAL_FILES_CHECKED"
+        echo "  Issues found: $TOTAL_ERRORS"
+        echo "  Warnings: $TOTAL_WARNINGS"
+
+    - name: Fail build on errors (optional)
+      if:
+        env.SHOULD_RUN == 'true' && env.HAS_MATCHING_CHANGES == 'true' &&
+        inputs.fail-on-errors == 'true'
+      shell: bash
+      run: |
+        set -euo pipefail
+        if [ "${TOTAL_ERRORS:-0}" -gt 0 ]; then
+          echo "::error::Failing because fail-on-errors=true and TOTAL_ERRORS=$TOTAL_ERRORS"
+          exit 1
```

Comment:
If this exits here the following steps won't run. If you want the following steps to run, you'll need an `if: always()`.